src/components/articles/ArticlesGrid.vue

```diff
@@ -0,0 +1,124 @@
+<template>
+  <main class="pa-2">
+    <v-row no-gutters>
+      <v-col v-for="(item, index) in items" :key="index" cols="12" sm="6" md="4" lg="3" xl="2">
+        <div class="pa-2">
+          <v-responsive :aspect-ratio="16 / 9">
+            <articles-article-small
+              @click="openDialog(index)"
+              :article="item.article"
+              :loading="item.loading"
+            />
+          </v-responsive>
+        </div>
+      </v-col>
+    </v-row>
+    <div v-intersect="{ handler: onIntersect, options: intersectOptions }"></div>
+    <articles-dialog v-if="!loading" ref="dialogRef" :items="items" />
+  </main>
+</template>
+
+<script lang="ts">
+import { defineComponent, computed, ref, Ref, watch } from '@vue/composition-api';
+import { CategoryEnum } from '@/config/categories/types';
+import ArticlesArticleSmall from './article/small/ArticlesArticleSmall.vue';
+import { useGetArticlesQuery, Article } from '@/generated/graphql';
+import ArticlesDialog from './ArticlesDialog.vue';
+
+const ITEMS_PER_PAGE = 64;
+type Item = {
+  article: Article;
+  loading: boolean;
+};
+
+export default defineComponent({
+  props: {
+    categoryId: String as () => CategoryEnum,
+  },
+  components: {
+    ArticlesArticleSmall,
+    ArticlesDialog,
+  },
+  setup(props) {
+    const numberOfPages = ref(1);
+    const numberOfItems = computed(() => numberOfPages.value * ITEMS_PER_PAGE);
+
+    // Infinite scroll
+    const needsMorePages = ref(null);
+    const intersectOptions = { rootMargin: '800px' };
```

Comment:
I think this margin should be somehow linked to the screen height. `800px` in a mobile device is a really short distance, since we scroll really fast there.